src/software/thunderscope/chicker/chicker.py

```diff
@@ -0,0 +1,186 @@
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import (
+    QApplication,
+    QCheckBox,
+    QGridLayout,
+    QGroupBox,
+    QButtonGroup,
+    QMenu,
+    QPushButton,
+    QRadioButton,
+    QVBoxLayout,
+    QHBoxLayout,
+    QWidget,
+    QSlider,
+)
+
+
+class ChickerWidget(QWidget):
+    def __init__(self, parent=None):
+        super(ChickerWidget, self).__init__(parent)
+
+        """
+        The Chicker Widget grid is laid out in the following way:
+                    ┌────────┐     ┌──────┐      ┌──────┐
+                    │ Charge │     │ Kick │      │ Chip │
+                    └────────┘     └──────┘      └──────┘
+
+                    ┌────────┐   ┌──────────┐ ┌─────────┐
+                    │No Auto │   │Auto Kick │ │Auto Chip│
+                    └────────┘   └──────────┘ └─────────┘
+
+                    ┌───────────────────────────────────┐
+                    │           Geneva Slider           │
+                    └───────────────────────────────────┘
+
+                    ┌───────────────────────────────────┐
+                    │           Power Slider            │
+                    └───────────────────────────────────┘
+        """
+        # grid --> groupBox --> button/slider
+        grid = QGridLayout()
+        self.radio_buttons = QButtonGroup()
+
+        self.charge_button_list = common_widgets.create_button("Charge", False)
+        self.charge_button = self.charge_button_list[1]
+        grid.addWidget(self.charge_button_list[0], 0, 0)
+
+        self.kick_button_list = common_widgets.create_button("Kick")
+        self.kick_button = self.kick_button_list[1]
+        grid.addWidget(self.kick_button_list[0], 0, 1)
+
+        self.chip_button_list = common_widgets.create_button("Chip")
+        self.chip_button = self.chip_button_list[1]
+        grid.addWidget(self.chip_button_list[0], 0, 2)
+
+        self.no_auto_list = common_widgets.create_radio("No Auto", self.radio_buttons)
+        self.no_auto_button = self.no_auto_list[1]
+        grid.addWidget(self.no_auto_list[0], 1, 0)
+
+        self.auto_kick_list = common_widgets.create_radio(
+            "Auto Kick", self.radio_buttons
+        )
+        self.auto_kick_button = self.auto_kick_list[1]
+        grid.addWidget(self.auto_kick_list[0], 1, 1)
+
+        self.auto_chip_list = common_widgets.create_radio(
+            "Auto Chip", self.radio_buttons
+        )
+        self.auto_chip_button = self.auto_chip_list[1]
+        grid.addWidget(self.auto_chip_list[0], 1, 2)
+
+        self.geneva_slider_list = common_widgets.create_slider("Geneva Slider", 1, 5, 1)
+        self.geneva_slider = self.geneva_slider_list[1]
+        grid.addWidget(self.geneva_slider_list[0], 2, 0, 1, 3)
+
+        self.power_slider_list = common_widgets.create_slider(
+            "Power Slider", 1, 100, 10
+        )
+        self.power_slider = self.power_slider_list[1]
+        grid.addWidget(self.power_slider_list[0], 3, 0, 1, 3)
```

Comment:
can the geneva and power slides have number labels and a textbox?

Reply:
I added a label that shows the current slider value. 
Having labels at each tick is not something that's built into pyqtgraph. Though it is possible and I tried it, but it gets far too complicated. [Here](https://gist.github.com/wiccy46/b7d8a1d57626a4ea40b19c5dbc5029ff) is an example.
I also tried adding a textbox, this wasn't too difficult but it made keeping track of the slider values complicated as all of the slider, label, and the textbox would need to be in sync. Also because the `create_slider` function was designed to be common for any other slider. Overall, it made the logic very complicated and code very non-readable, hence I decided to go against including it.

I would happily be able to include these features if we really need to. Let me know!